gossip/privdata/pvtdataprovider.go

```diff
@@ -202,27 +202,44 @@ func (pdp *PvtdataProvider) RetrievePvtdata(pvtdataToRetrieve []*ledger.TxPvtdat
 
 	pvtdata := make(rwsetByKeys)
 
+	//If there is no private data to retrieve for the block, skip all population attempts and return
+	if len(pvtdataRetrievalInfo.eligibleMissingKeys) == 0 {
+		pdp.logger.Debugf("No eligible collection private write sets to fetch for block [%d]", pdp.blockNum)
+		retrievedPvtdata.pvtdataRetrievalInfo = pvtdataRetrievalInfo
+		retrievedPvtdata.blockPvtdata = pdp.prepareBlockPvtdata(pvtdata, pvtdataRetrievalInfo)
+		return retrievedPvtdata, nil
+	}
+
+	numEligibleMissingKeysToRetrieve := len(pvtdataRetrievalInfo.eligibleMissingKeys)
+
 	// POPULATE FROM CACHE
+	numRemainingToFetch := len(pvtdataRetrievalInfo.eligibleMissingKeys)
 	pdp.populateFromCache(pvtdata, pvtdataRetrievalInfo, pvtdataToRetrieve)
+	numFetchedFromLocalCache := numRemainingToFetch - len(pvtdataRetrievalInfo.eligibleMissingKeys)
```

Comment:
Not related to this PR. Calling it as a cache is confusing. It just refers to the pvtdata passed along with the block. Couldn't think of an alternative name too.